internal/common/disasters/setup.go

```diff
@@ -0,0 +1,22 @@
+package disasters
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/config"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+var envConf = config.GameConfig().DisasterConfig
+
+// InitEnvironment initialises environment according to definitions
+func InitEnvironment(islandIDs []shared.ClientID) *Environment {
+
+	ag := ArchipelagoGeography{islands: map[shared.ClientID]Island{}, xBounds: envConf.XBounds, yBounds: envConf.YBounds}
+	dp := disasterParameters{globalProb: envConf.GlobalProb, spatialPDF: envConf.SpatialPDFType, magnitudeLambda: envConf.MagnitudeLambda}
+
+	for i, id := range islandIDs {
+		island := Island{id, float64(i), float64(0)} // begin with equidistant points on x axis
+		ag.islands[id] = island
+	}
+	//TODO: think about possible security concerns of returning a pointer
+	return &Environment{geography: ag, disasterParams: dp, lastDisasterReport: DisasterReport{}} // returning a pointer so that other methods can modify returned Environment instance
```

Comment:
structs are reference types in golang, there's no need to return a pointer for saving memory FYI.

Actually i just realised you're doing it for modification.
I'd prefer it to be entirely replaced when changed, i.e.

```
e := s.gameState.Environment
// do whatever in e
s.gameState.Environment = e
```

Reply:
Cool, got you